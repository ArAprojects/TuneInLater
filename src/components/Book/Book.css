.Book_section {
  border: 3px solid black;
  background-color: lightgrey;
  height: 150px;
  width: 70%;
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  padding: 10px;
  margin: 10px;
}

.Book_section p {
  width: 200px;
}

.Book_section h3 {
  width: 200px;
}

.Book_image {
  height: 150px;
}

.Book_image-favorite {
  position: absolute;
  top: 7px;
  right: 10px;
  height: 40px;
}

.Book_image-favorite:hover {
  transform: scale(1.2) perspective(100px);
  transition: all 0.5s ease;
}

h3 {
  text-align: center;
}

.Book_button {
  height: 35px;
  width: 120px;
  padding: 5px;
  font-size: 14px;
  background-color: #142431;
  color: white;
  font-weight: bold;
  opacity: 0.8;
  position: absolute;
  bottom: 10px;
  right: 10px;
}

.Book_p-favorites {
  position: absolute;
  top: 20px;
  right: 9px;
  font-size: 12px;
}
