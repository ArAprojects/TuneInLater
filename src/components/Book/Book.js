import React from 'react';
import { Link } from 'react-router-dom';
import headphones from './headphones.svg';
import headphonesActive from './headphones-active.svg'
import './Book.css';

const Book = ({ book_name, author_name, artwork_url, book_id, primary_genre_name, release_date, description, toggleFavorite, favorite }) => {

  return(
    <section className="Book_section">
      <button onClick={() => toggleFavorite({ book_name, author_name, artwork_url, book_id, description, primary_genre_name, release_date, favorite }) } className="Book_button">Star</button>
      <h3>{book_name}</h3>
      <p>{author_name}</p>
      <img src={artwork_url} alt={book_name}/>
      <img src={headphones}/>
      <img src={headphonesActive}/>
      <Link to={`/details/${book_id}`} className="link">
        <button>See more Info</button>
      </Link>
    </section>
  )
}

export default Book;
