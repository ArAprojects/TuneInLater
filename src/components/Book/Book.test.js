import React from 'react';
import { shallow } from 'enzyme';
import Book from './Book';

describe('Book', () => {
  let wrapper;
  const mockToggleFavorite = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Book
      key={1}
      book_name='The Cat in the Hat'
      book_id={1}
      author_name='Dr. Seuss'
      artwork_url='http//:image.png'
      toggleFavorite={mockToggleFavorite}
      favorite={false}
      description='Story Description'
      primary_genre_name='children'
      release_date='04/23/1990'
    />)
  })

  it('should keep an up-to-date snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
