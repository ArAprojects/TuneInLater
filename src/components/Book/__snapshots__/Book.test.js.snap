// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should keep an up-to-date snapshot 1`] = `
<section
  className="Book_section"
>
  <button
    className="Book_button"
    onClick={[Function]}
  >
    Star
  </button>
  <h3>
    The Cat in the Hat
  </h3>
  <p>
    Dr. Seuss
  </p>
  <img
    alt="The Cat in the Hat"
    src="http//:image.png"
  />
  <Link
    className="link"
    to="/details/1"
  >
    <button>
      See more Info
    </button>
  </Link>
</section>
`;
